src/main/kotlin/me/zeroeightsix/kami/feature/module/MurderMysteries.kt

```diff
@@ -0,0 +1,124 @@
+    private var updateEvery: @Constrain.Range(min = 2.0, max = 16.0, step = Double.MIN_VALUE) Int = 16
+
+    private var lastUpdate: Long = 0L
+
+    private fun murdererIs(username: String, item: String): MutableText? {
+        lastUpdate = System.currentTimeMillis()
+        return Texts.f(Formatting.WHITE, Texts.append(
+                Texts.lit("[MurderMysteries] "),
+                Texts.flit(Formatting.AQUA, username),
+                Texts.flit(Formatting.WHITE, " is the "),
+                Texts.flit(Formatting.RED, "murderer! "),
+                Texts.flit(Formatting.WHITE, "Wields "),
+                Texts.flit(Formatting.GOLD, "$item!")
+        ))
+    }
+    private fun detectiveIs(username: String, item: String): MutableText? {
+        lastUpdate = System.currentTimeMillis()
+        return Texts.f(Formatting.WHITE, Texts.append(
+                Texts.lit("[MurderMysteries] "),
+                Texts.flit(Formatting.AQUA, username),
+                Texts.flit(Formatting.WHITE, " has "),
+                Texts.flit(Formatting.GOLD, "$item!")
+        ))
+    }
+
+
+    @EventHandler
+    val worldListener = Listener<TickEvent.Client.InGame>({
+        if ((lastUpdate + updateEvery * 1000L <= System.currentTimeMillis()))
+            mc.world?.players?.forEach { player ->
+                val username = player.displayName.string
+
+                //Make sure the player is not being told that they themself are murderer
+                if (username == mc.player?.displayName?.string) return@forEach
+
+                //Check for weaponly items (swords, axes)
+                player.itemsEquipped.forEach { it ->
+                    if (weaponItems) when (it.item) {
+                        Items.NETHERITE_SWORD -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.DIAMOND_SWORD -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.IRON_SWORD -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.GOLDEN_SWORD -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.WOODEN_SWORD -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+
+                        Items.NETHERITE_AXE -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.DIAMOND_AXE -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.IRON_AXE -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.GOLDEN_AXE -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                        Items.WOODEN_AXE -> mc.player?.sendMessage(murdererIs(username, it.item.name.string), false)
+                    }
```

Comment:
Instead of checking for the individual items, just check if `it.item is AxeItem || it.item is SwordItem`